### PR TITLE
migrate limits & overrides to sqlx / async, introduce async test wrapper

### PR DESCRIPTION
**Pull request by syphar**
This migrates limits & overrides to sqlx / async. 
Also this is a first idea of an `async_wrapper` for tests. 

I have some other ideas how to improve the test setup further, but since test database creation / cleanup is still sync, it's harder in this PR, and I would move the task to a later point.

### CODE REVIEW

File: src/docbuilder/limits.rs [hunk 1]
```diff
@@ -109,78 +114,88 @@ mod test {
...
-        });
+        })
+        .await;
     }
 
-    #[test]
-    fn overrides_dont_lower_memory_limit() {
-        wrapper(|env| {
-            let db = env.db();
+    #[tokio::test]
```
Comment (Nemo157) [inline src/docbuilder/limits.rs:176]: I haven't looked at the rest yet, but one thing I dislike about using `#[tokio::test]` is that we then aren't using the `ctx.runtime()`. We have some code like `utils::spawn_blocking` that relies on the ambient runtime while other code gets it directly from the context, so there may be some weird effects from having these different.

Comment (syphar) [reply src/docbuilder/limits.rs:176]: That's definitely a valid point. 

What do you think is the better goal to have? Using the contrext runtime? Or tokio::main and tokio::test?

Comment (Nemo157) [reply src/docbuilder/limits.rs:176]: If we can get rid of `Context::runtime` and ensure everywhere has an ambient runtime available, that's probably the best long-term plan. That has issues around `postgres` and `reqwest::blocking` code not being able to run anywhere within a runtime though.

Comment (syphar) [reply src/docbuilder/limits.rs:176]: So this feels like we should keep `tokio::test` here, while preventing that there is a second runtime. Probably `ctx.runtime()` just returning the ambient runtime and never creating a new one? 

The short term alternative for this PR would probably be to make the test methods sync again, also make the `async_wrapper` sync, while internally using the ctx.runtime to `block_on`.

Comment (syphar) [reply src/docbuilder/limits.rs]: I chose the simple approach in this simple first version, to not block this PR further. 

We won't start using `tokio::test` yet, I'll look into this in a later PR again.

File: src/test/mod.rs [hunk 2]
```diff
@@ -448,6 +485,27 @@ impl TestEnvironment {
         })
     }
 
+    pub(crate) async fn async_db(&self) -> &TestDatabase {
+        let config = self.config().clone();
+        let runtime = self.runtime().clone();
+        let instance_metrics = self.instance_metrics().clone();
+        if self.db.get().is_none() {
```
Comment (Nemo157) [inline src/test/mod.rs:492]: We could maybe switch `self.db` to a `tokio::sync::OnceCell`, and have `db()` just do `self.runtime().block_on(self.async_db())`.

**Review by Nemo157 — CHANGES_REQUESTED** [review]

File: src/test/mod.rs (hunk 2 above)
Comment (syphar) [reply src/test/mod.rs]: good idea! looks much nicer (apart from the remainin `spawn_blocking` ;) )

**Comment by syphar** [comment]
will fix the test failure and re-request the review

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**